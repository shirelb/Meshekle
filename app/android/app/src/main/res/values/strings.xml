<resources>
    <string name="app_name">משקלה</string>
    <string name="default_notification_channel_id" translatable="false">fcm_default_channel</string>
</resources>
