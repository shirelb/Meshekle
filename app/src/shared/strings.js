const loginScreenStrings = {
    LOGIN: "התחבר",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainScreenStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
};

const drawerMenu = {
    LOGOUT: "התנתק",
    MAIN_SCREEN_NAME: "בית",
    APPOINTMENTS_SCREEN_NAME: "תורים",
    CHORES_SCREEN_NAME: "תורנויות",
    PhoneBook_SCREEN_NAME: "ספר טלפונים",
};

export default {loginScreenStrings,mainScreenStrings,drawerMenu};