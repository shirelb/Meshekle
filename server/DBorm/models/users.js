module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};