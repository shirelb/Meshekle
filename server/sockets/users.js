
module.exports = function (socket,pushToAllWebClient,pushToAllAppClient) {
    socket.on('userShallowDeleted', (data) => {
        console.log('socket userShallowDeleted data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userCreated', (data) => {
        console.log('socket userCreated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });

    socket.on('userUpdated', (data) => {
        console.log('socket userUpdated data ', data);
        pushToAllWebClient('getUsers');
        pushToAllAppClient('getUsers');
    });


};
