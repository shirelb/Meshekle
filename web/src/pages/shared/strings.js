const loginPageStrings = {
    LOGIN: "התחבר",
    SUBMIT: "שלח",
    WELCOME_TO_LOGIN: "Welcome to the login screen!",
    USER_ID_PLACEHOLDER: "ת.ז.",
    PASSWORD_PLACEHOLDER: "סיסמא",
    APP_NAME: "משקל\'ה",
    WRONG_CREDENTIALS: "אחד הפרטים שהזנת לא נכון",
    NO_SUCH_USER: "המשתמש אינו קיים",
    EMPTY_USER_ID: "הוסף ת.ז. ונסה להתחבר שוב",
    EMPTY_PASSWORD: "הוסף סיסמא ונסה להתחבר שוב",
};

const mainScreenStrings = {
    LOGOUT: "התנתק",
    WELCOME: "ברוכים הבאים ",
};

const drawerMenu = {
    LOGOUT: "התנתק",
    MainScreenName: "בית",
    AppointmentsScreenName: "תורים",
    ChoresScreenName: "תורנויות",
};

export default {loginPageStrings, mainScreenStrings, drawerMenu};