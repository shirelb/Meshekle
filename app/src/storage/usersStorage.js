import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
