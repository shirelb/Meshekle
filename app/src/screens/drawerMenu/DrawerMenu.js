import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {DrawerActions, NavigationActions} from 'react-navigation';
import {View} from 'react-native';
import phoneStorage from "react-native-simple-store";
import {Avatar, Icon, List, ListItem} from 'react-native-elements'
import strings from '../../shared/strings'
import UserProfileInfo from "../../components/userProfile/UserProfileInfo";
import usersStorage from "../../storage/usersStorage";
import UserProfileForm from "../../components/userProfile/UserProfileForm";
import {APP_SOCKET} from "../../shared/constants";

export default class DrawerMenu extends Component {
    constructor(props) {
        super(props);

        this.state = {
            formModal: false,
            infoModal: false,
            userLoggedin: {},
        };
    }

    navigateToScreen = (route) => () => {
        const navigateAction = NavigationActions.navigate({
            routeName: route
        });
        this.props.navigation.dispatch(navigateAction);
        this.props.navigation.dispatch(DrawerActions.closeDrawer())
    };

    onLogoutPress = () => {
        phoneStorage.update('userData', {
            token: null,
            userId: null,
            userFullname: null
        })
            .then(() => {
                APP_SOCKET.emit('disconnectAppClient', {userId: this.userId});
                this.props.navigation.navigate('Auth');
            })
    };

    componentWillMount() {
        phoneStorage.get('userData')
            .then(userData => {
                // console.log('agenda componentDidMount userData ', userData);
                this.userHeaders = {
                    'Authorization': 'Bearer ' + userData.token
                };
                this.userId = userData.userId;
                this.loadUser();
            });
    }

    componentDidMount() {
        APP_SOCKET.on("getUsers", this.loadUser.bind(this));
    }

    componentWillUnmount() {
        APP_SOCKET.off("getUsers");
    }

    loadUser() {
        usersStorage.getUserById(this.userId, this.userHeaders)
            .then(user => {
                console.log("DrawerMenu user ", user);
                let userLoggedin = user.data[0];

                this.setState({
                    userLoggedin: userLoggedin,
                    formModal: false,
                    infoModal: false,
                });
            })
    };

    componentWillReceiveProps(nextProps, nextContext) {
        this.setState({
            formModal: false,
            infoModal: false,
            // userLoggedin: {},
        });
    }

    setFormModalVisible = () => {
        this.setState({
            formModal: true,
            infoModal: false,
            // userLoggedin: {},
        });
    }

    render() {
        return (
            <View style={{flex: 1, backgroundColor: '#ededed'}}>
                <List containerStyle={{marginBottom: 20}}>
                    <ListItem
                        containerStyle={{height: 150}}
                        roundAvatar
                        onPress={() => this.setState({infoModal: true})}
                        avatar={<Avatar
                            large
                            rounded
                            source={{uri: this.state.userLoggedin.image}}
                            activeOpacity={0.7}
                        />}
                        // key={i}
                        title={this.state.userLoggedin.fullname}
                        subtitle="צפייה בפרופיל"
                        // rightIcon=null
                        hideChevron
                        titleStyle={{fontSize: 20}}
                    />
                    <ListItem
                        // roundAvatar
                        onPress={this.navigateToScreen('MainScreen')}
                        // avatar={l.avatar_url}
                        // key={i}
                        title={strings.drawerMenu.MAIN_SCREEN_NAME}
                        // subtitle="test"
                        leftIcon={{name: 'home'}}
                        rightIcon={<Icon name={'chevron-left'}/>}
                        // badge={{ value: 3, textStyle: { color: 'orange' } }}
                    />
                    <ListItem
                        // roundAvatar
                        onPress={this.navigateToScreen('AppointmentsScreen')}
                        // avatar={l.avatar_url}
                        // key={i}
                        title={strings.drawerMenu.APPOINTMENTS_SCREEN_NAME}
                        // subtitle="test"
                        leftIcon={{name: 'insert-invitation'}}
                        rightIcon={<Icon name={'chevron-left'}/>}
                        badge={{value: 3, textStyle: {color: 'orange'}}}
                    />
                    <ListItem
                        // roundAvatar
                        onPress={this.navigateToScreen('ChoresScreen')}
                        // avatar={l.avatar_url}
                        // key={i}
                        title={strings.drawerMenu.CHORES_SCREEN_NAME}
                        // subtitle="test"
                        leftIcon={{name: 'transfer-within-a-station'}}
                        rightIcon={<Icon name={'chevron-left'}/>}
                        badge={{value: 3, textStyle: {color: 'orange'}}}
                    />
                    <ListItem
                        // roundAvatar
                        onPress={this.navigateToScreen('PhoneBookScreen')}
                        // avatar={l.avatar_url}
                        // key={i}
                        title={strings.drawerMenu.PhoneBook_SCREEN_NAME}
                        // subtitle="test"
                        leftIcon={{name: 'contacts'}}
                        rightIcon={<Icon name={'chevron-left'}/>}
                        badge={{value: 3, textStyle: {color: 'orange'}}}
                    />
                    <ListItem
                        // roundAvatar
                        onPress={this.onLogoutPress.bind(this)}
                        // avatar={l.avatar_url}
                        // key={i}
                        title={strings.drawerMenu.LOGOUT}
                        // subtitle="test"
                        leftIcon={{name: 'power'}}
                        // rightIcon=null
                        hideChevron
                    />
                </List>

                <UserProfileInfo
                    modalVisible={this.state.infoModal}
                    user={this.state.userLoggedin}
                    openedFrom={"DrawerMenu"}
                    userHeaders={this.userHeaders}
                    setFormModalVisible={this.setFormModalVisible}
                />

                <UserProfileForm
                    modalVisible={this.state.formModal}
                    user={this.state.userLoggedin}
                    userHeaders={this.userHeaders}
                    loadUser={this.loadUser.bind(this)}
                />

            </View>
        );
    }
}

DrawerMenu.propTypes = {
    navigation: PropTypes.object
};