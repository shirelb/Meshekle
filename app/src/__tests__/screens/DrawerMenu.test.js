import React from 'react';

import {shallow} from "enzyme/build";
import DrawerMenu from "../../screens/drawerMenu/DrawerMenu";
import {DrawerActions, NavigationActions} from 'react-navigation';
import strings from "../../shared/strings";
import usersStorage from "../../storage/usersStorage";
import phoneStorage from "react-native-simple-store";
import users from "../jsons/users";

jest.mock("../../storage/serviceProvidersStorage");
jest.mock("../../storage/usersStorage");
jest.mock("react-native-simple-store");
jest.mock("react-navigation");


describe("DrawerMenu should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {};
    const userTest = users[2];
    const userIdTest = "972350803";
    const userFullnameTest = "Dion Revance";
    const mockStore = {
        userData: {
            serviceProviderId: "549963652",
            userId: "549963652",
            token: "some token"
        }
    };
    const navigation = {navigate: jest.fn(), dispatch: jest.fn()};

    usersStorage.getUsers = jest.fn().mockResolvedValue(users);
    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    phoneStorage.update = jest.fn().mockImplementation((key, value) => Promise.resolve(mockStore[key] = value));

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('match snapshot', async () => {
        wrapper = shallow(<DrawerMenu navigation={navigation}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("render what the user see", async () => {
        wrapper = await shallow(<DrawerMenu navigation={navigation}/>);
        componentInstance = wrapper.instance();

        await wrapper.update();

        expect(wrapper.find('List')).toHaveLength(1);
        expect(wrapper.find('ListItem')).toHaveLength(8);
        expect(wrapper.find('ListItem').at(1).props().title).toEqual(strings.drawerMenu.MAIN_SCREEN_NAME);
        expect(wrapper.find('ListItem').at(2).props().title).toEqual(strings.drawerMenu.APPOINTMENTS_SCREEN_NAME);
        expect(wrapper.find('ListItem').at(3).props().title).toEqual(strings.drawerMenu.CHORES_SCREEN_NAME);
        expect(wrapper.find('ListItem').at(4).props().title).toEqual(strings.drawerMenu.PhoneBook_SCREEN_NAME);
        expect(wrapper.find('ListItem').at(5).props().title).toEqual(strings.drawerMenu.ANNOUNCEMENTS_SCREEN_NAME);
        expect(wrapper.find('ListItem').at(6).props().title).toEqual(strings.drawerMenu.REPORT);
        expect(wrapper.find('ListItem').at(7).props().title).toEqual(strings.drawerMenu.LOGOUT);
        expect(wrapper.find('UserProfileInfo')).toHaveLength(0);
        expect(wrapper.find('UserProfileForm')).toHaveLength(0);

        expect(componentInstance.state.formModal).toEqual(false);
        expect(componentInstance.state.infoModal).toEqual(false);
        expect(componentInstance.state.userLoggedin).toEqual({});
    });

    it("handle setFormModalVisible", async () => {
        wrapper = shallow(<DrawerMenu navigation={navigation}/>);
        componentInstance = wrapper.instance();

        expect(componentInstance.state.formModal).toEqual(false);
        expect(componentInstance.state.infoModal).toEqual(false);

        await componentInstance.setFormModalVisible();
        expect(componentInstance.state.formModal).toEqual(true);
        expect(componentInstance.state.infoModal).toEqual(false);
    });

    it("handle onLogoutPress", async () => {
        wrapper = shallow(<DrawerMenu navigation={navigation}/>);
        componentInstance = wrapper.instance();

        await componentInstance.onLogoutPress();
        expect(componentInstance.props.navigation.navigate).toHaveBeenCalled();
    });

    xit("navigate To Screen", async () => {
        NavigationActions.navigate = jest.fn();
        DrawerActions.closeDrawer.jest.fn();

        wrapper = shallow(<DrawerMenu navigation={navigation}/>);
        componentInstance = wrapper.instance();

        // await componentInstance.navigateToScreen('AppointmentsScreen');
        await wrapper.find('ListItem').at(2).props().onPress('AppointmentsScreen');
        expect(NavigationActions.navigate).toHaveBeenCalledTimes(1);
        expect(componentInstance.props.navigation.dispatch).toHaveBeenCalledTimes(2);
    });

});
