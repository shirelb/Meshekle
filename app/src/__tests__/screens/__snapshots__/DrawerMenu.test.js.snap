// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenu should match snapshot 1`] = `
<ScrollViewMock
  style={
    Object {
      "backgroundColor": "#ededed",
      "flex": 1,
    }
  }
>
  <List
    containerStyle={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <ListItem
      avatar={
        <Avatar
          activeOpacity={0.7}
          editButton={
            Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            }
          }
          large={true}
          onEditPress={null}
          rounded={true}
          showEditButton={false}
          source={
            Object {
              "uri": undefined,
            }
          }
        />
      }
      chevronColor="#bdc6cf"
      containerStyle={
        Object {
          "height": 150,
        }
      }
      disabled={false}
      hideChevron={true}
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={true}
      subtitle="צפייה בפרופיל"
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      titleNumberOfLines={1}
      titleStyle={
        Object {
          "fontSize": 20,
        }
      }
      underlayColor="white"
    />
    <ListItem
      badge={
        Object {
          "containerStyle": Object {
            "backgroundColor": "transparent",
            "opacity": 0,
          },
          "textStyle": Object {
            "color": "#00adf5",
          },
          "value": "👋",
        }
      }
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      leftIcon={
        Object {
          "name": "home",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        <Icon
          color="black"
          name="chevron-left"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="בית"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      badge={
        Object {
          "containerStyle": Object {
            "backgroundColor": "transparent",
            "opacity": 0,
          },
          "textStyle": Object {
            "color": "#00adf5",
          },
          "value": "👋",
        }
      }
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      leftIcon={
        Object {
          "name": "insert-invitation",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        <Icon
          color="black"
          name="chevron-left"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="תורים"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      badge={
        Object {
          "containerStyle": Object {
            "backgroundColor": "transparent",
            "opacity": 0,
          },
          "textStyle": Object {
            "color": "#00adf5",
          },
          "value": "👋",
        }
      }
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      leftIcon={
        Object {
          "name": "transfer-within-a-station",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        <Icon
          color="black"
          name="chevron-left"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="תורנויות"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      badge={
        Object {
          "containerStyle": Object {
            "backgroundColor": "transparent",
            "opacity": 0,
          },
          "textStyle": Object {
            "color": "#00adf5",
          },
          "value": "👋",
        }
      }
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      leftIcon={
        Object {
          "name": "contacts",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        <Icon
          color="black"
          name="chevron-left"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="ספר טלפונים"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      badge={
        Object {
          "containerStyle": Object {
            "backgroundColor": "transparent",
            "opacity": 0,
          },
          "textStyle": Object {
            "color": "#00adf5",
          },
          "value": "👋",
        }
      }
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={false}
      leftIcon={
        Object {
          "name": "insert-comment",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        <Icon
          color="black"
          name="chevron-left"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={24}
          underlayColor="white"
        />
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="לוח מודעות"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={true}
      leftIcon={
        Object {
          "name": "error",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="דווח על תקלה"
      titleNumberOfLines={1}
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      disabled={false}
      hideChevron={true}
      leftIcon={
        Object {
          "name": "power",
        }
      }
      leftIconUnderlayColor="white"
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      rightTitleNumberOfLines={1}
      roundAvatar={false}
      subtitleNumberOfLines={1}
      switchButton={false}
      textInputEditable={true}
      title="התנתק"
      titleNumberOfLines={1}
      underlayColor="white"
    />
  </List>
</ScrollViewMock>
`;
